### PR TITLE
Sdlog2 posix

### PR DESCRIPTION
**Pull request by tumbili**
@mcharleb @LorenzMeier These changes make the sdlog2 module work for posix. I have not looked yet why the code which I commented out gives a floating point error. I will look into it and see what the problem is. I'm not too sure about the  way the access function is supposed to be used, so review would be nice.

### CODE REVIEW

File: src/drivers/device/vdev_posix.cpp [hunk 1]
```diff
@@ -270,7 +270,7 @@ int px4_fsync(int fd)
 
 int px4_access(const char *pathname, int mode)
 {
-	if (mode == F_OK) {
```
Comment (tumbili) [inline src/drivers/device/vdev_posix.cpp]: @mcharleb This cannot work for the orb_exists() function because we are calling it with F_OK.

**Comment by tumbili** [comment]
This is an example log:
http://logs.uaventure.com/view/UWVQhBvPYTkNRNHwWTDM3D

**Comment by tumbili** [comment]
@LorenzMeier @mcharleb The floating point exception is not also fixed. This is good to go from my side.

**Comment by LorenzMeier** [comment]
Thanks!

File: src/modules/systemlib/perf_counter.c [hunk 2]
```diff
@@ -396,13 +396,12 @@ perf_print_counter_fd(int fd, perf_counter_t handle)
 	case PC_ELAPSED: {
 		struct perf_ctr_elapsed *pce = (struct perf_ctr_elapsed *)handle;
 		float rms = sqrtf(pce->M2 / (pce->event_count-1));
```
Comment (mcharleb) [inline src/modules/systemlib/perf_counter.c]: @tumbili @LorenzMeier Is is possible that pce->event_count could be == 1 here?

File: src/drivers/device/vdev_posix.cpp (hunk 1 above)
Comment (mcharleb) [inline src/drivers/device/vdev_posix.cpp]: @tumbili Thanks. No idea what I was thinking.

**Pull request merged**